### PR TITLE
Market/config upgrade

### PR DESCRIPTION
**Pull request by tworec**
Why:
- sync config changes from `master` to `release/v0.6`
- enable users to configure cleanup intervals (will help @cryptobench)

What:
- cherry-pick commit from #1304 which was targeting `master`
- make cleanup interval configurable
- move two other env vars to the config

Relates to #453 and #460

### CODE REVIEW

File: core/market/src/config.rs [hunk 1]
```diff
@@ -33,37 +35,61 @@ pub struct SubscriptionConfig {
     pub default_ttl: chrono::Duration,
 }
 
-#[derive(Clone)]
+#[derive(StructOpt, Clone)]
 pub struct EventsConfig {
+    #[structopt(env = "MAX_MARKET_EVENTS_DEFAULT", default_value = "20")]
```
Comment (jiivan) [inline core/market/src/config.rs:40]:
```suggestion
    #[structopt(env = "MARKET_MAX_EVENTS_DEFAULT", default_value = "20")]
```

File: core/market/src/config.rs [hunk 2]
```diff
@@ -33,37 +35,61 @@ pub struct SubscriptionConfig {
...
     pub max_events_default: i32,
+    #[structopt(env = "MAX_MARKET_EVENTS_MAX", default_value = "100")]
```
Comment (jiivan) [inline core/market/src/config.rs:42]:
```suggestion
    #[structopt(env = "MARKET_MAX_EVENTS_MAX", default_value = "100")]
```

File: core/market/src/config.rs [hunk 3]
```diff
@@ -33,37 +35,61 @@ pub struct SubscriptionConfig {
...
     pub max_events_max: i32,
 }
 
+#[derive(StructOpt, Clone)]
+pub struct DbConfig {
+    /// Interval in which Market cleaner will be invoked
+    #[structopt(env = "MARKET_DB_CLEANUP_INTERVAL", parse(try_from_str = humantime::parse_duration), default_value = "24h")]
+    pub cleanup_interval: Duration,
+    /// Number of days to persist Agreements and related Agreement Events
+    #[structopt(env = "MARKET_AGREEMENT_STORE_DAYS", default_value = "90")]
```
Comment (jiivan) [inline core/market/src/config.rs:52]: Why drop the `YAGNA_` prefix?

Comment (jiivan) [reply core/market/src/config.rs:52]: Besides the default, there was also the `min` value (30). Will it be ignored now?

**Comment by tworec** [comment]
I've cherrypicked #1292 to fix flaky tests.

File: core/market/src/config.rs (hunk 3 above)
Comment (tworec) [reply core/market/src/config.rs:52]: > Why drop the `YAGNA_` prefix?

To be consistent with the rest of env vars here.

> Besides the default, there was also the `min` value (30). Will it be ignored now?

I was struggling with this decision but finally decided to drop this feature.

**Review by jiivan — APPROVED** [review]

**Pull request merged**